pkg/core/interop_neo.go

```diff
@@ -231,48 +232,23 @@ func createContractStateFromVM(ic *interop.Context, v *vm.VM) (*state.Contract,
 	if len(script) > MaxContractScriptSize {
 		return nil, errors.New("the script is too big")
 	}
-	paramBytes := v.Estack().Pop().Bytes()
-	if len(paramBytes) > MaxContractParametersNum {
-		return nil, errors.New("too many parameters for a script")
-	}
-	paramList := make([]smartcontract.ParamType, len(paramBytes))
-	for k, v := range paramBytes {
-		paramList[k] = smartcontract.ParamType(v)
-	}
-	retType := smartcontract.ParamType(v.Estack().Pop().BigInt().Int64())
-	properties := smartcontract.PropertyState(v.Estack().Pop().BigInt().Int64())
-	name := v.Estack().Pop().Bytes()
-	if len(name) > MaxContractStringLen {
-		return nil, errors.New("too big name")
-	}
-	version := v.Estack().Pop().Bytes()
-	if len(version) > MaxContractStringLen {
-		return nil, errors.New("too big version")
-	}
-	author := v.Estack().Pop().Bytes()
-	if len(author) > MaxContractStringLen {
-		return nil, errors.New("too big author")
-	}
-	email := v.Estack().Pop().Bytes()
-	if len(email) > MaxContractStringLen {
-		return nil, errors.New("too big email")
-	}
-	desc := v.Estack().Pop().Bytes()
-	if len(desc) > MaxContractDescriptionLen {
-		return nil, errors.New("too big description")
-	}
-	contract := &state.Contract{
-		Script:      script,
-		ParamList:   paramList,
-		ReturnType:  retType,
-		Properties:  properties,
-		Name:        string(name),
-		CodeVersion: string(version),
-		Author:      string(author),
-		Email:       string(email),
-		Description: string(desc),
-	}
-	return contract, nil
+	manifestBytes := v.Estack().Pop().Bytes()
+	if len(manifestBytes) > manifest.MaxManifestSize {
+		return nil, errors.New("manifest is too big")
+	}
+	if !v.AddGas(util.Fixed8(interopGasRatio * (len(script) + len(manifestBytes)))) {
+		return nil, errors.New("gas limit exceeded")
+	}
+	var m manifest.Manifest
+	r := io.NewBinReaderFromBuf(manifestBytes)
+	m.DecodeBinary(r)
+	if r.Err != nil {
+		return nil, r.Err
+	}
+	return &state.Contract{
```

Comment:
And what's the ID for the new contract?

Reply:
Forget this part. It is set differently in `contractCreate` and `contractMigrate`